src/main.ts

```diff
@@ -16,8 +16,7 @@ export async function main() {
     const CONTAINER_NAME = `MICROSOFT_AZURE_CLI_${getCurrentTime()}_CONTAINER`;
     try {
         if (process.env.RUNNER_OS != 'Linux') {
-            core.setFailed('Please use Linux based OS as a runner.');

```

Comment:
According to Git Blame, this line was added as early as https://github.com/Azure/cli/commit/9932b03c5ec88c4638b554ef50643f0804c7b08f. Later, https://github.com/Azure/cli/commit/548c3ec692451811afd4a68a596eec087ecfc434 added `src/entrypoint.ts` which overrides `core.setFailed` with https://github.com/Azure/cli/blob/548c3ec692451811afd4a68a596eec087ecfc434/src/entrypoint.ts#L6